tools/autograd/derivatives.yaml

```diff
@@ -1467,7 +1467,15 @@
 
 # We never call _linalg_svd with compute_uv=False in an autograd context, so we don't even consider it here
 - name: _linalg_svd(Tensor A, bool full_matrices=False, bool compute_uv=True) -> (Tensor U, Tensor S, Tensor Vh)
-  A: svd_backward(grads, full_matrices, U, S, Vh)
+  A: "svd_backward(full_matrices && grad_U.defined() ? grad_U.narrow(-1, 0, S.size(-1)) : grad_U,
+                   grad_S,
+                   full_matrices && grad_Vh.defined() ? grad_Vh.narrow(-2, 0, S.size(-1)) : grad_Vh,
+                   full_matrices ? U.narrow(-1, 0, S.size(-1)) : U,
+                   S,
+                   full_matrices ? Vh.narrow(-2, 0, S.size(-1)) : Vh)"
```

Comment:
the narrowing is no longer done in the backward?

Reply:
I moved it here to remove the `full_matrices` keyword from the backwards. `svd_backwards` is used in a number of places, and I wanted to make sure no one could call it with `full_matrices=True`, as the usual caveat applies: svd with `full_matrices=True` is not a well-defined function so, in particular, it is not differentiable.